zebra-consensus/src/checkpoint.rs

```diff
@@ -939,11 +975,28 @@ where
             // so we don't need to panic here. The persistent state is correct even when the
             // task is cancelled, because block data is committed inside transactions, in
             // height order.
-            if zebra_chain::shutdown::is_shutting_down() {
+            let result = if zebra_chain::shutdown::is_shutting_down() {
                 Err(VerifyCheckpointError::ShuttingDown)
             } else {
                 result.expect("commit_finalized_block should not panic")
+            };
+            if result.is_err() {
+                // If there was an error comitting the block, then this verifier
+                // will be out of sync with the state. In that case, reset
+                // its progress back to the state tip.
+                let tip = match state_service
+                    .oneshot(zs::Request::Tip)
+                    .await
+                    .map_err(Into::into)
+                    .map_err(VerifyCheckpointError::Tip)?
+                {
+                    zs::Response::Tip(tip) => tip,
+                    _ => unreachable!("wrong response for Tip"),
+                };
+                // TODO: do we need to handle errors?
```

Comment:
`send` only errors when we drop the checkpoint verifier:
> An unsuccessful send would be one where the corresponding receiver has already been deallocated

https://doc.rust-lang.org/nightly/std/sync/mpsc/struct.Sender.html#method.send

So we can just ignore errors here - we only drop the verifier when we're shutting down.
(And even if we did clean it up earlier, we don't care about any resets once it's dropped.)